src/main.c

```diff
@@ -61,12 +62,23 @@ int main(void)
             newDirection.x = 1;
             newDirection.y = 0;
             break;
+        case 27:
+            quitPending = 1;
+            break;
         }
 
         {
             int shouldQuit = querySignals(term);
             if (shouldQuit)
             {
+                clearScreen(&screen);
+                printf("You quit. SCORE: %d\n", game.scoring);
+                break;
+            }
+            if (quitPending)
+            {
+                clearScreen(&screen);
+                printf("You quit. SCORE: %d\n", game.scoring);
```

Comment:
Please merge the two `if`s into one to avoid repeating the body
For example, do `if (shouldQuit || quitPending)`